tests/unit/lms/services/jstor/factory_test.py

```diff
@@ -7,9 +7,44 @@
 
 
 class TestServiceFactory:
-    def test_it(
-        self, pyramid_request, application_instance_service, http_service, JSTORService
+    def test_it(self, pyramid_request, http_service, JSTORService):
+        svc = service_factory(sentinel.context, pyramid_request)
+
+        JSTORService.assert_called_once_with(
+            api_url=sentinel.jstor_api_url,
+            secret=sentinel.jstor_api_secret,
+            enabled=sentinel.jstor_enabled,
+            site_code=sentinel.jstor_site_code,
+            http_service=http_service,
+            tracking_user_agent="Chrome 100",
+            tracking_user_id=pyramid_request.lti_user.h_user.username,
+        )
+        assert svc == JSTORService.return_value
+
+    def test_it_omits_tracking_info_if_not_available(
```

Comment:
The main `test_it` test has tracking info enabled because that's the normal case. This is for the exceptional scenario that the JSTOR service is used in some context where there is no LTI user or User-Agent header.